packages/pages/src/common/src/function/types.ts

```diff
@@ -0,0 +1,131 @@
+import { GetPath } from "../template/types.js";
+import { PluginEvent } from "../ci/ci.js";
+
+/**
+ * Defines the functions and fields that are available to the serverless function.
+ * @public
+ */
+export interface ServerlessFunctionModule<
+  U extends ServerlessFunctionArgument
+> {
+  /** The exported config function */
+  config?: ServerlessFunctionConfig;
+  /** The exported getPath function */
+  getPath?: GetPath<void>;
+  /** The exported function */
+  default: ExecuteServerlessFunction<U>;
+}
+
+/**
+ * The type definition for serverless function itself.
+ * @public
+ */
+export type ExecuteServerlessFunction<T extends ServerlessFunctionArgument> = (
+  props: T
+) => ServerlessFunctionReturnValue;
+
+/**
+ * The exported `config` function's definition.
+ * @public
+ */
+export interface ServerlessFunctionConfig {
+  /** The name of the serverless function feature. */
+  name?: string;
+  /** The http event. */
+  event?: PluginEvent;
```

Comment:
We don't need this either after all. It will always default to API for serverless. Keep in mind we still want this in the internal type though.